src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncContext.Unix.cs

```diff
@@ -489,6 +711,53 @@ public override void InvokeCallback(bool allowPooling)
 
                 cb(bt, sa, sal, rf, ec);
             }
+
+            internal override unsafe bool TryBatch(SocketAsyncContext context, int id, ref Interop.Sys.IoControlBlock ioControlBlock)
+            {
+                if (Buffer.Length == 0 || IsSync || !TrySetRunning())
```

Comment:
important: `Buffer.Length == 0` - AIO does not support performing a zero read operations, which is commonly used by ASP.NET to check whether there is some data available  for read (now this is controllable by using https://github.com/dotnet/aspnetcore/pull/19396)